runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -439,14 +440,14 @@ MM_CopyForwardScheme::updateLeafRegions(MM_EnvironmentVLHGC *env)
 				Assert_MM_true(updatedSpineRegion->containsObjects());
 
 				/* we need to move the leaf to another region's leaf list since its spine has moved */
-				region->_allocateData.removeFromArrayletLeafList();
+				region->_allocateData.removeFromArrayletLeafList(env);
 				region->_allocateData.addToArrayletLeafList(updatedSpineRegion);
```

Comment:
consider removing this pass for offheap. basically 
keep a pool of ARRAYLET_RESERVED regions, and iterate only them to convert them to FREE
if array surivived don't do anything (by not setting spine pointer in region descriptor during allocation)